### PR TITLE
Create Python SDK Distroless variant

### PR DESCRIPTION
**Pull request by damondouglas**
Addresses https://github.com/apache/beam/issues/32817 adding Python SDK (non-ML) [distroless](https://github.com/GoogleContainerTools/distroless) variant. **Future PR will add a GitHub workflow to run checks**.

Before submitting this PR, the following tests passed:

```
:sdks:python:test-suites:dataflow:py39:validatesDistrolessContainer
:sdks:python:test-suites:dataflow:py310:validatesDistrolessContainer
:sdks:python:test-suites:dataflow:py311:validatesDistrolessContainer
:sdks:python:test-suites:dataflow:py312:validatesDistrolessContainer
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by shunping** [comment]
Thanks! Could you also trigger the workflow of beam_PostCommit_Python? Trying to make sure it won't break that post commit again. ;)

File: sdks/python/container/Dockerfile-distroless [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+COPY --from=base /usr/local/bin /usr/local/bin
+
+# Contains the Python library dependencies.
+COPY --from=base /usr/local/lib /usr/local/lib
+
+# Python standard library modules.
+COPY --from=base /usr/lib/python* /usr/lib/.
+
+# Contains the boot entrypoint and related files such as licenses.
+COPY --from=base /opt /opt
```
Comment (liferoad) [inline sdks/python/container/Dockerfile-distroless:41]: ENV LANG C.UTF-8

File: sdks/python/container/Dockerfile-distroless [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+ENV PATH "$PATH:/usr/local/bin"
+
+# Despite the ENTRYPOINT set in base image, need to reset since deriving the layer derives from a different image.
+ENTRYPOINT ["/opt/apache/beam/boot"]
```
Comment (liferoad) [inline sdks/python/container/Dockerfile-distroless:46]: COPY --from=build-env /usr/lib/locale /usr/lib/locale

**Comment by damondouglas** [comment]
R: @Abacn

**Review by liferoad — APPROVED** [review]

File: sdks/python/test-suites/dataflow/common.gradle [hunk 3]
```diff
@@ -380,6 +380,50 @@ task validatesContainer() {
...
+      ]
+    }
+    exec {
+      executable 'docker'
+      args = ['push', imageURL]
+    }
+    exec {
+      def testTarget = "apache_beam/examples/wordcount_it_test.py::WordCountIT::test_wordcount_it"
+      def argMap = [
+              "output"             : "gs://temp-storage-for-end-to-end-tests/py-it-cloud/output",
```
Comment (Abacn) [inline sdks/python/test-suites/dataflow/common.gradle:410]: ideally the temp location and project to run the test should be configurable. However I checked in Java SDK's gradle files it is the case, but not for python SDK's gradle files, so it is fine for now

**Review by Abacn — APPROVED** [review]

**Pull request merged**